qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -249,38 +249,25 @@ def _compute_weighted_kernel_sum(self, index: int, X: np.ndarray, training: bool
         Returns:
             Weighted sum of kernel evaluations employed in the Pegasos algorithm
         """
-        kernel: Dict[Tuple, np.ndarray] = {}
+        # non-zero indices corresponding to the support vectors
+        support_vectors = list(self._alphas.keys())
+
+        # for training
+        if training:
+            x = X[support_vectors]
+        # for prediction
+        else:
+            x = self._x_train[support_vectors]
+        if not self._precomputed:
+            # evaluate kernel function only for the fixed datum and the support vectors
+            kernel = self._quantum_kernel.evaluate(X[index], x) + self._kernel_offset
+        else:
+            kernel = X[index, support_vectors]
+
+        y = np.array(list(map(self._label_map.get, self._y_train[support_vectors])))
+        a = np.array(list(self._alphas.values()))
```

Comment:
Could you please either rename `x`, `y` and `a` to something more meaningful or add comments to the code what they mean?

Reply:
Thanks for the comment, I (hopefully) clarified the names. And yes, the change seems trivial. However, evaluating the kernel entries as a batch is much more efficient compared to doing it over a loop due to how QuantumKernel is optimized, so I think the change is justified.